data-prepper-plugins/aws-lambda/src/main/java/org/opensearch/dataprepper/plugins/lambda/common/LambdaCommonHandler.java

```diff
@@ -58,4 +44,40 @@ public void waitForFutures(List<CompletableFuture<Void>> futureList) {
             }
         }
     }
+
+    public static List<Buffer> createBufferBatches(Collection<Record<Event>> records,
+                                                   String keyName,
+                                                   String whenCondition,
+                                                   ExpressionEvaluator expressionEvaluator,
+                                                   int maxEvents,
+                                                   ByteCount maxBytes,
+                                                   Duration maxCollectionDuration,
+                                                   List<Record<Event>> resultRecords) {
+        // Initialize here to void multi-threading issues
+        // Note: By default, one instance of processor is created across threads.
+        Buffer currentBufferPerBatch = new InMemoryBuffer(keyName);
+        List<Buffer> batchedBuffers = new ArrayList<>();
+
+        LOG.info("Batch size received to lambda processor: {}", records.size());
+        for (Record<Event> record : records) {
+            final Event event = record.getData();
+
+            // If the condition is false, add the event to resultRecords as-is
+            if (whenCondition != null && !expressionEvaluator.evaluateConditional(whenCondition, event)) {
```

Comment:
When condition is not applicable to sink. I guess you want to send null for sinks.